### PR TITLE
Fix #48 add ecp arguments to main function and cli parser

### LINKED ISSUE #48: Add ecp arguments to main() function and CLI parser
- include the `ecp` arguments to the CLI parser (for both SPAHM-a and SPAHM-b)
- include a check into main() to set `ecp` when `minao` is used and `args.elements`  
  contains atoms with Z>39

### PR DESCRIPTION
**Pull request by YAY-C**
- **Add ECP argument to CLI**
- **Add check to set ecp when necessary with 'minao' basis**
- **Added ecp argument to CLI parser and check for minao basis**

### CODE REVIEW

File: qstack/spahm/rho/bond.py [hunk 1]
```diff
@@ -78,6 +79,13 @@ def main():
     if args.print>0: print(vars(args))
     correct_num_threads()
 
+    import pyscf.data.elements as pysymb
```
Comment (briling) [inline qstack/spahm/rho/bond.py:82]: * I don't think it's good to forcely set ecp. Better raise an error
* This checking code should be a separate function in `compound.py`, if not inside of `xyz_to_mol`

**Review by briling — CHANGES_REQUESTED** [review]

**Comment by YAY-C** [comment]
Should we make a test for `xyz_to_mol()` (397b2f4)?

_if yes, would the test `assert()` the `RuntimeError` raising?_

**Comment by briling** [comment]
> Should we make a test for `xyz_to_mol()` ([397b2f4](https://github.com/lcmd-epfl/Q-stack/commit/397b2f48b4ddb6108497cf989f3e98c593f8815f))?
> 
> _if yes, would the test `assert()` the `RuntimeError` raising?_

There's a test in `test_compound.py` but it doesn't really do anything. Can be updated in another batch of changes

File: qstack/compound.py [hunk 2]
```diff
@@ -50,10 +50,14 @@ def xyz_to_mol(fin, basis="def2-svp", charge=0, spin=0, ignore=False, unit='ANG'
         mol.charge = -(sum(map(data.elements.charge, elements))%2)
         mol.spin = 0
 
-    if ecp:
+    if ecp != None:
```
Comment (briling) [inline qstack/compound.py:53]: I think `is not None`/`is None` (below) is more clean

File: qstack/compound.py [hunk 3]
```diff
@@ -50,10 +50,14 @@ def xyz_to_mol(fin, basis="def2-svp", charge=0, spin=0, ignore=False, unit='ANG'
...
         mol.ecp = ecp
 
     mol.build()
+    species_charges = [data.elements.charge(z) for z in mol.elements]
+    if mol.basis == 'minao' and ecp == None and (np.array(species_charges) > 39).any():
```
Comment (briling) [inline qstack/compound.py:58]: minao needs ecp for Y (hence `>=39`) but I think it should be `>36` (after Kr)

**Review by briling — CHANGES_REQUESTED** [review]

**Review by YAY-C — COMMENTED** [review]
- for the `is None`/`is not None`
I checked and there are a few of those in other files (not concerned by this PR `kernel_utils.py`, `hyperparameters.py`, `rep_completion.py`).
Maybe it's better to make this changes in another batch.
_I still have two PR with different changes to request and one of them includes changes in theses files!_

File: qstack/compound.py [hunk 4]
```diff
@@ -50,12 +50,12 @@ def xyz_to_mol(fin, basis="def2-svp", charge=0, spin=0, ignore=False, unit='ANG'
...
 
-    if ecp != None:
+    if ecp is not None:
         mol.ecp = ecp
 
     mol.build()
     species_charges = [data.elements.charge(z) for z in mol.elements]
-    if mol.basis == 'minao' and ecp == None and (np.array(species_charges) > 39).any():
+    if mol.basis == 'minao' and ecp is None and (np.array(species_charges) > 36).any():
         msg = f"{mol.basis} basis set requires the use of effective core potentials for atoms with Z>39"
```
Comment (briling) [inline qstack/compound.py:59]: also here! sorry

**Comment by briling** [comment]
Okay let's fix is/== None another time!

**Review by briling — CHANGES_REQUESTED** [review]

**Pull request merged**